src/axes/axis.ts

```diff
@@ -150,9 +178,271 @@ export class Axis<D> extends Component {
+        visibility: visibilityF
+      });
+
+    bindElements(this._annotationContainer.select(".annotation-circle-container"), "circle", "annotation-circle")
+      .attr({
+        cx: isHorizontal ? positionF : secondaryPosition,
+        cy: isHorizontal ? secondaryPosition : positionF,
+        r: 3
+      });
+
+    let rectangleOffsetF = (d: D) => {
+      switch (this.orientation()) {
+        case "bottom":
+        case "right":
+          return offsetF(d);
+        case "top":
+        case "left":
+          return offsetF(d) - measurements.get(d).height;
+      }
+    };
+    bindElements(this._annotationContainer.select(".annotation-rect-container"), "rect", "annotation-rect")
+      .attr({
+        x: isHorizontal ? positionF : rectangleOffsetF,
+        y: isHorizontal ? rectangleOffsetF : positionF,
+        width: isHorizontal ? (d) => measurements.get(d).width : (d) => measurements.get(d).height,
+        height: isHorizontal ? (d) => measurements.get(d).height : (d) => measurements.get(d).width,
+        visibility: visibilityF
+      });
+
+    let annotationWriter = this._annotationWriter;
+    let annotationFormatter = this.annotationFormatter();
+    let annotationLabels = bindElements(this._annotationContainer.select(".annotation-label-container"), "g", "annotation-label");
+    annotationLabels.selectAll(".text-container").remove();
+    annotationLabels.attr({
+        transform: (d) => {
+          let xTranslate = isHorizontal ? positionF(d) : rectangleOffsetF(d);
+          let yTranslate = isHorizontal ? rectangleOffsetF(d) : positionF(d);
+          return `translate(${xTranslate},${yTranslate})`;
+        },
+        visibility: visibilityF
+      })
+      .each(function (annotationLabel) {
+        let writeOptions = {
+          selection: d3.select(this),
+          xAlign: "center",
+          yAlign: "center",
+          textRotation: isHorizontal ? 0 : 90
+        };
+        annotationWriter.write(annotationFormatter(annotationLabel),
+                                 isHorizontal ? measurements.get(annotationLabel).width : measurements.get(annotationLabel).height,
+                                 isHorizontal ? measurements.get(annotationLabel).height : measurements.get(annotationLabel).width,
+                                 writeOptions);
+      });
+  }
+
+  private _annotatedTicksInDomain() {
+    let scaleRange = this._scale.range();
+    return this.annotatedTicks().filter((tick) => {
+      if (tick == null) {
+        return false;
+      }
+      return Utils.Math.inRange(this._scale.scale(tick), scaleRange[0], scaleRange[1]);
+    });
+  }
+
+  /**
+   * Retrieves the height of the core pieces of the Axis,
+   * where height is vertical for horizontal Axes and horizontal for vertical Axes.
+   *
+   * The core pieces include the axis labels, the end tick marks, the inner tick marks, and the tick label padding.
+   */
+  protected _coreAxisHeight() {
```

Comment:
"`_coreSize()`"? We know it's on `Axis`, and "height" could be confusing given the existence of `height()` (which *is* direction-dependent).